pkg/master/manager.go

```diff
@@ -0,0 +1,167 @@
+package master
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/go-ping/ping"
+	"github.com/medik8s/self-node-remediation/pkg/peers"
+	corev1 "k8s.io/api/core/v1"
+	"os/exec"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	initErrorText                = "error initializing master handler"
+	kubeletPort                  = "10250"
+	endpointToAccess             = "www.google.com"
+)
+
+var (
+	initError    = errors.New(initErrorText)
+	processError = errors.New("an error occurred during master remediation process")
+)
+
+//Manager contains logic and info needed to fence and remediate master nodes
+type Manager struct {
+	nodeName                             string
+	nodeRole                             peers.Role
+	wasEndpointAccessibleAtStart         bool
+	client                               client.Client
+	log                                  logr.Logger
+}
+
+//NewManager inits a new Manager return nil if init fails
+func NewManager(nodeName string, myClient client.Client) *Manager {
+	return &Manager{
+		nodeName:                     nodeName,
+		client:                       myClient,
+		wasEndpointAccessibleAtStart: false,
+		log:                          ctrl.Log.WithName("master").WithName("Manager"),
+	}
+}
+
+func (manager *Manager) Start(ctx context.Context) error {
+	if err := manager.initializeManager(); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Did this work in a manual test?

AFAIK you need to block before returning, otherwise the controller manager will stop. See doc of method specification.
You can block with:

```
	select {
	case <-ctx.Done():
	}
```

Reply:
Yep it worked.
```
// It's very important that Start blocks until
// it's done running.
```
I think the reason for that is that  `manager.initializeManager()` is synced - it just sets some fields and done, it is not suppose to set up any async processes.

Reply:
Oh, indeed, interesting. Seems the main usecase of the context is cleanup on shutdown (which includes stopping async processes).
Wondering why I had issues with a similar case without async code in the past though 🤔